test/kitchen/docs/README.md

```diff
@@ -1,42 +1,128 @@
+
+The `.gitlab-ci.yml` file has been structured in a way to avoid config duplication:
+
+- a common `kitchen_common` job from which all kitchen testing jobs inherit
+- `kitchen_agent_aX` jobs that specify the Agent version tested
+- `kitchen_os_X` jobs that specify which OS is tested
+- `kitchen_test_X` jobs that specify which test suite is run
+- `kitchen_datadog_X_flavor` jobs that specify which Agent flavor is run (eg. `datadog-agent`, `datadog-iot-agent`)
+- `kitchen_azure_location_X` jobs that specify the Azure location where the tests are run.
+
+**Note:** We spread our kitchen tests on multiple regions mainly because we have a limited number of resources available on each region on Azure.
+
+From these base blocks, we define:
+- test types: `kitchen_test_<test_suite>_<flavor>` which is a combination of a chosen test suite, Agent flavor and Azure location
+- scenarios: `kitchen_scenario_<OS>_aX` which is a combination of a chosen OS and major Agent version.
+
+**Note:** To avoid having too many different test types, we bind each test suite with a specific Azure location. Eg. all install script tests are run in the same Azure location.
+
+The real jobs that are run in the Gitlab pipelines are thus a combination of test types and scenarios, named `kitchen_<OS>_<test_suite>_<flavor>-aX`.
+
+##### testkitchen_cleanup
+
+The `testkitchen_cleanup_azure-aX` jobs clean up the Azure VMs that were used during the `testkitchen_testing` stage.
+
+**Note:** The kitchen test jobs send a `destroy` request to Azure when their tests end, so usually these jobs do nothing. They're mainly useful to make sure all VMs have been destroyed, or when kitchen jobs are cancelled while they're running.
+
+**Note:** Azure VMs are also cleaned up on a daily basis by a job in the `mars-jenkins-scripts` pipeline.
+
+The `testkitchen_cleanup_s3-aX` jobs clean up the S3 buckets of the testing repos.
+
+**Important note:** These jobs are always run, even when a kitchen test fails. This is done on purpose (to avoid having remaining resources once a job fails).
+The downside of this approach is that when you want to retry a failing ktichen test job, you'll have to re-run the corresponding deploy job first (to put the Agent packages back in the testing repository).
+
+#### Test suites
+
+kitchen tests are defined in a file named `kitchen.yml` which defines:
+- the provisioner (`provisioner:` key) used in the tests. In the pipeline, `chef-solo` is used.
+- the driver (`driver:` + `driver_config:` keys) used to run the tests. In the pipeline, the [Azure kitchen driver](https://github.com/test-kitchen/kitchen-azurerm) is used.
+
+- the list of platforms (`platforms:` key) used in the tests.
+- the list of test suites (`suites:` key) to run.
+
+To reduce duplication, the `kitchen.yml` file is generated at test time by the `run-test-kitchen.sh` script ([see below](#test-execution-tasks/run-test-kitchen.sh)) by concatenating 2 template files:
```

Comment:
It seems like Github removes special characters from ids so the link didn't work
```suggestion
To reduce duplication, the `kitchen.yml` file is generated at test time by the `run-test-kitchen.sh` script ([see below](#test-execution-tasksrun-test-kitchensh)) by concatenating 2 template files:
```